### PR TITLE
Simplify Array trait type

### PR DESCRIPTION
**Pull request by SylvainCorlay**
- [x] make `to_json` and `from_json` free functions and pass them to `tag` in the marks

### CODE REVIEW

File: bqplot/traits.py [hunk 1]
```diff
@@ -98,110 +84,83 @@ def from_json(value, obj=None):
...
         if (dim > 1) and (1 in shape):
-            value = np.squeeze(value) if (self.squeeze) else value
+            value = np.squeeze(value) if self.squeeze else value
             dim = len(np.shape(value))
         if self.allow_none and dim == 0:
             return value
+        # validate dimension
         if (dim > max_dim or dim < min_dim):
-            raise TraitError("Dimension mismatch")
+            raise TraitError('Dimension mismatch: ' + str(dim))
```
Comment (dmadeka) [inline bqplot/traits.py]: Shouldnt we add more to this?

`Dimension mismatch: 0` doesn't help a lot no?

Comment (SylvainCorlay) [inline bqplot/traits.py]: ah that is a debug thing. there was no intension of adding it.

**Comment by SylvainCorlay** [comment]
I have another PR coming on the top of this one that makes dtype and order kwargs instead of metada and remove the call to tag so that you have to call `.tag(sync=True, **array_serialization)`, but this first step should not be disruptive.

**Comment by SylvainCorlay** [comment]
Closing for another coming PR.

**Pull request closed without merging**